### PR TITLE
Add meta-documentation

### PR DESCRIPTION
**Pull request by maliesa96**

### CODE REVIEW

File: docs/user/writing_documentation.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+Do keep your doc:
+ - Self-contained - avoid having users go through multiple pages to understand how to use a feature or become familiar with a workflow.
+ -  Accessible - Include examples to help demonstrate a point or feature, and clarify code snippets by providing context.
+ - Readable - Make use of markdown styling (see [this page]([https://commonmark.org/help/](https://commonmark.org/help/)) for a markdown cheatsheet) - see the section below for more on styling.
+
+ ### Don'ts:
+
+ - Claim other work as your own. Please cite external sources - you may use footnotes for this purpose.
+ - Shy away from pointing to external resources. There are several high-quality online resources that explain various reinforcement learning techniques. We refer users to them all the time, you should too!
```
Comment (krzentner) [inline docs/user/writing_documentation.md:27]: This is really a "Do" and would be clearer in that section.

File: docs/user/writing_documentation.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+High-level documentation should focus on describing such workflows, or, in the case of newly-added features, describing how these features might be used when running experiments. For code-level documentation such as docstrings and style guides, please see the [CONTRIBUTING.md](https://github.com/rlworkgroup/garage/blob/master/CONTRIBUTING.md).
+
+
+## Where to Put Documentation
+You can choose to update the existing docs under the `docs/user` directory, or create a new page dedicated to the subject you are documenting.
+
+In certain cases, it may be beneficial to include your documentation in a README file in a directory specific to that new feature or workflow. Garage's `docker` folder, for example, contains the main Dockerfile and `entrypoint.sh` scripts that are used to run the docker builds. This folder contains a separate README with details on garage's docker images and how to run them.
```
Comment (krzentner) [inline docs/user/writing_documentation.md:11]: I think we've decided that actually these README's should just point to a docs page in `docs/user`.

File: docs/user/writing_documentation.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+def very_readable(int arg):
+```
+You can also use one bat tick for inline code snippets, class references, or directory paths:
+
+
+``you can `cd`  into `~/garage/data/` `` :   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; you can `cd` into `~/garage/data/`
+
+
+### Sphinx
+
```
Comment (ryanjulian) [inline docs/user/writing_documentation.md:53]: maybe include a math directive to demonstrate you can render math in these docs too?

File: docs/user/writing_documentation.md (hunk 2 above)
Comment (ryanjulian) [reply docs/user/writing_documentation.md:11]: agreed -- we will be moving this to docs/ as well

File: docs/user/writing_documentation.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# Writing Documentation
+When adding new features to garage, it is important that these features are well-documented and accessible. Garage's high-level docs contain information on workflows and processes that are regularly used by developers and end-users. They are home to various how-to's and examples that help others use garage effectively.
+
+High-level documentation should focus on describing such workflows, or, in the case of newly-added features, describing how these features might be used when running experiments. For code-level documentation such as docstrings and style guides, please see the [CONTRIBUTING.md](https://github.com/rlworkgroup/garage/blob/master/CONTRIBUTING.md).
+
+
+## Where to Put Documentation
+You can choose to update the existing docs under the `docs/user` directory, or create a new page dedicated to the subject you are documenting.
```
Comment (ryanjulian) [inline docs/user/writing_documentation.md:9]: don't you need to update the table of contents in `docs/index.rst`?

File: docs/user/writing_documentation.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+```python
+def very_readable(int arg):
+```
+You can also use one bat tick for inline code snippets, class references, or directory paths:
+
+
+``you can `cd`  into `~/garage/data/` `` :   &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; you can `cd` into `~/garage/data/`
+
+
+### Sphinx
```
Comment (ryanjulian) [inline docs/user/writing_documentation.md:52]: i think you should cover how to view your docs locally (you can run `make docs`) and open `docs/_build/html/index.html` to see exactly what ReadTheDocs will render.

you will also have to note that you need `garage[dev]` installed to do this.

**Comment by ryanjulian** [comment]
Update `docs/index.rst`?

**Comment by ryanjulian** [comment]
I just found this resource which would probably be helpful to mention https://www.writethedocs.org/guide/

**Comment by ryanjulian** [comment]
Sorry for all the quick changes -- I pushed a reorg of the TOC so you will have to update the section on adding the docs to the TOC.

I also decided that all documentation authors should get acknowledged in a section at the bottom, so please add that too.

See https://github.com/rlworkgroup/garage/pull/1638 for an example.

Don't forget to acknowledge yourself here too!

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by maliesa96** [comment]
@Mergifyio rebase

**Comment by maliesa96** [comment]
closes #1547

**Comment by maliesa96** [comment]
@ahtsan can you a take a look at this pls

**Comment by maliesa96** [comment]
@mergifyio rebase

**Comment by maliesa96** [comment]
@ryanjulian can you force push this? I fixed a typo in a docstring so the CI will run the whole test suite.

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Pull request merged**